pipeline/DESIGN.md

```diff
@@ -0,0 +1,190 @@
+
+TBD how this table is populated and its schema.
+This table may be able to be combined with the streams table. Should we?
+
+## event_states
+
+The event_states table contains a row for each event in a stream and the state of the document at that point in the stream.
+
+### Features
+
+* Access to the full history of states for a stream
+* Validation of the model schema
+
+### Schema
+
+| Column      | Type              | Description                                                                                     |
+| ------      | ----              | -----------                                                                                     |
+| index       | u64               | Order of this event. Index is always greater than the index of any previous event in the stream |
+| stream_cid  | bytes             | Cid of the stream                                                                               |
+| stream_type | u8                | Stream type                                                                                     |
+| controller  | string            | Controller of the stream                                                                        |
+| dimensions  | map(string,bytes) | Set of key values dimension pairs of the stream                                                 |
+| event_cid   | bytes             | Cid of the event                                                                                |
+| event_type  | u8                | One of Data (0) or Time (1)                                                                     |
+| state       | string            | JSON encoding of the event state                                                                |
+
+### Transformation
+
+This table computes the aggregated state for each conclusion event.
+Additionally it validates the aggregated state matches the model schema of the stream.
+
+## stream_heads
+
+The stream_states table contains a row for head of each stream representing the canonical state of the stream heads.
+
+### Features
+
+* Access to the multiple heads of streams for users building their on conflict resolution
+
+### Schema
+
+| Column      | Type              | Description                                                                                     |
+| ------      | ----              | -----------                                                                                     |
+| index       | u64               | Order of this event. Index is always greater than the index of any previous event in the stream |
+| stream_cid  | bytes             | Cid of the stream                                                                               |
+| stream_type | u8                | Stream type                                                                                     |
+| controller  | string            | Controller of the stream                                                                        |
```

Comment:
maybe we can add something about controller typically being a did:key or did:pkh somewhere in this doc?